### DISCUSSION TITLE
The basic constructure of Backend project

### DISCUSSION THREAD

**Discussion by BaikSeung** [mmuhunn/CampusShare#18] [General]
# Project Features and Data Structure

### 1. **User Data**
- `name`
- `userId` (student number)
- `password`
- `rank`

### 2. **Sharing Data**
- `uploadId`
- `studentNumber`
- `name`
- `title`
- `content`
- `file`
- `numberOfDownloads`
- `uploadTime` (timestamp)

### 3. **Question Data**
- `questionId`
- `authorName`
- `authorId`
- `content` (question)
- `answerData`

### 4. **Answer Data**
- `answerId`
- `questionId`
- `authorName`
- `authorId`
- `content`

---

### **1. Login Feature**
- **Sign Up**: Users can sign up using `name`, `studentNumber`, and `password`.
- **Login**: Users log in with `ID` (student number) and `password`.  
  (*Note: Remember your password!*)

### **2. Sharing Feature**
- **Upload**:  
  Requires the following mandatory fields:  
  - `uploadId`  
  - `studentNumber`  
  - `name`  
  - `title`  
  - `content`  
  - `file`  
  Additionally, the `uploadTime` (timestamp) is recorded.  

- **View All**:  
  Displays all shared content.  

- **Search**:  
  Search shared content by `title`.  

- **Download**:  
  Increments the `numberOfDownloads` counter.

### **3. Q&A Board**
- The **Sharing Feature** applies here, excluding the file attachment.  
- Users can:  
  - Ask questions  
  - Provide answers  

---

## Implementation Order
1. Login functionality  
2. Sharing feature  
3. Community feature (Q&A board)  
4. Ranking system  
5. Course and professor evaluations  

---

If you have any suggestions or improvements, please feel free to share!

**Comment by mmuhunn** [comment]
Good organize.👍
However, I think Question Data and Answer Data also need time stamp datas.
